OpenSim/Actuators/MuscleFixedWidthPennationModel.h

```diff
@@ -342,19 +325,27 @@ OpenSim_DECLARE_CONCRETE_OBJECT(MuscleFixedWidthPennationModel, Object);
                              double muscleVelocity,
                              double tendonVelocity) const;
 
-    void ensureModelUpToDate();
+protected:
+    // Component interface.
+    void extendFinalizeFromProperties() override;
 
 private:
     void setNull();
     void constructProperties();
-    void buildModel();
-    void ensurePropertiesSet() const;
 
     double m_parallelogramHeight;
     double m_maximumSinPennation;
     double m_minimumFiberLength;
     double m_minimumFiberLengthAlongTendon;
 
+    // Enforces a lower bound on the fiber length to avoid a numerical
+    // singularity as the fiber length approaches zero.
+    double clampFiberLength(double fiberLength) const;
+
+    // These classes are friends because they call clampFiberLength().
+    friend class Thelen2003Muscle;
+    friend class Millard2012EquilibriumMuscle;
+    friend class Millard2012AccelerationMuscle;
```

Comment:
>Why do the consumers of the PennationModel have to call clampFiberLength on the PennationModel?

See [this line](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Actuators/Thelen2003Muscle.cpp#L302), for example.